tests/test_scatterPlot.py

```diff
@@ -117,11 +121,40 @@ def test_ScatterPlotWithTwoHistsTask(self):
                                bands=self.bands,
                                plotName="test")
 
+        self.assertTrue(isinstance(result.scatterPlot, plt.Figure))
+
         filename_figure_tmp = os.path.join(self.testDir, "test_scatterPlot.png")
+        texts, lines = get_and_remove_figure_text(result.scatterPlot)
+
+        # Set to true to re-generate reference data
+        resave = False
+
+        for idx, line in enumerate(lines):
+            filename = os.path.join(path_lines_ref, f"line_{idx}.txt")
+            if resave:
+                np.savetxt(filename, line)
+            arr = np.loadtxt(filename)
+            assert np.isclose(arr, line).flat
+
+        filename_texts_tmp = os.path.join(self.testDir, "test_scatterPlot_texts.txt")
+        filenames = [filename_texts_tmp]
+        if resave:
+            filenames.append(filename_texts_ref)
+
+        for filename in filenames:
+            with open(filename, 'w') as f:
+                f.writelines(f'{text}\n' for text in texts)
+
+        self.assertTrue(filecmp.cmp(filename_texts_tmp, filename_texts_ref))
```

Comment:
I think we could make this check a bit more robust to minor changes by replacing the full-file comparisons with logic that checks that the new plot's text lines (as a `set` of `str` lines) are a superset of those saved in the file.  That would let us choose save just the lines we think are particularly important (e.g. by git interactive staging after running with `resave=True`), and more importantly it would insulate us from matplotlib deciding to order the lines differently in response to e.g. figure elements being moved around or added to the figure in a different order.